### PR TITLE
Use golang container in tests

### PR DESCRIPTION
**Pull request by vdo**
Use golang base container as a environment for tests, instead of using the host machine

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -25,6 +21,7 @@ jobs:
         go test -vet=off -timeout=15m -race -covermode atomic -coverprofile=covprofile ./...
 
     - name: Go analyze
+      shell: bash
```
Comment (mvdan) [inline .github/workflows/main.yml]: can we just make bash the default shell for the entire job?

**Review by mvdan — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (vdo) [reply .github/workflows/main.yml]: shell is a property of run, not of the job, as per https://json.schemastore.org/github-workflow

Comment (mvdan) [reply .github/workflows/main.yml]: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_iddefaultsrun

Comment (vdo) [reply .github/workflows/main.yml]: Great, I will use this default instead then

**Review by p4u — APPROVED** [review]

**Comment by p4u** [comment]
Merge it yourself @vdo

**Pull request merged**